sw/device/tests/sim_dv/sram_ctrl_ret_scrambled_access_test.c

```diff
@@ -102,15 +106,15 @@ static void test_ram_write_read_pattern(void) {
  */
 static void test_ram_scrambling(const dif_sram_ctrl_t *sram_ctrl) {
   // Write the pattern at the start of RAM.
-  sram_ctrl_testutils_write(kRetSramStartAddr, &kRamTestPattern1);
+  sram_ctrl_testutils_write(
+      kRetSramStartAddr,
+      (sram_ctrl_testutils_data_t){.words = kRamTestPattern1,
+                                   .len = ARRAYSIZE(kRamTestPattern1)});
 
   sram_ctrl_testutils_scramble(sram_ctrl);
+  CHECK_ARRAYS_NE((uint32_t *)kRetSramStartAddr, kRamTestPattern1,
+                  ARRAYSIZE(kRamTestPattern1));
 
-  mmio_region_t region = mmio_region_from_addr(kRetSramStartAddr);
-  for (int i = 0; i < SRAM_CTRL_TESTUTILS_DATA_NUM_WORDS; ++i) {
-    uint32_t read_data = mmio_region_read32(region, sizeof(uint32_t) * i);
-    CHECK(read_data != kRamTestPattern1.words[i]);
```

Comment:
This loop used to check that all words were different. Might be a significant different for scrambling tests. `read_from_ram_check` seems to support this behavior.

Reply:
I changed to reproduce the same behavior by coping to a tmp buffer, so that will generate the correct amount of ECC errors.